pages/blog/building-async-flight-notification-service.md

```diff
@@ -0,0 +1,257 @@
+    example: "4" 
+```
+
+Messages are shared among services so it’s important to correctly organize the YAML definition files under a common folder. In our case, we call it common:
+
+        common/
+            messages/
+                flight_queue.yaml
+                flight_status.yaml
+            schemas/
+                flight.yaml
+                segment.yaml
+                user.yaml
+
+Services communicate through channels using the publish/subscribe pattern. Our architecture uses two different channels:
+
+- `flight/queue` to manage and queue the flights to be monitored.
+- `flight/update` to manage the notifications about flight updates.
+
+Each service contains an `asyncapi.yaml` file with the description of the service and server and channel information. Let's take a look to the final `asyncapi.yaml` file of the Subscriber service to see how the messages and channels are organized:
+
+```yaml
+asyncapi: '2.0.0'
+info:
+  title: Flight Subscriber Service
+  version: '1.0.0'
+  description: |
+     Allows users to subscribe events from a given flight
+  license:
+    name: Apache 2.0
+    url: 'https://www.apache.org/licenses/LICENSE-2.0'
+servers:
+  development:
+    url: mqtt://localhost:1883
+    protocol: mqtt
+channels:
+  flight/queue:
+    description: |
+      queue flight in order to retrieve status
+    subscribe:
+      summary: Receive information about the flight that should be monitored for changes
+      message:
+        $ref: '#/components/messages/flightQueue'
+components:
+  messages:
+    flightQueue:
+      $ref: '../common/messages/flight_queue.yaml'
+```
+
+When the user provides their flight information, the Subscriber service emits a `flightQueue` message that will be received by the Monitor service from the `flight/queue` channel. The Notifier service also receives the message and adds the payload to the list of flights to monitor.
+
+Once the Monitor service detects a change in flight status (e.g. a change in boarding gate), it emits a `flightStatus` message to inform subscribers. The Notifier service, which is subscribed to the changes on the `flight/update` channel, notifies the end-user by SMS.
+
+The AsyncAPI specification files for the [Monitor Service](https://github.com/amadeus4dev/amadeus-async-flight-status/blob/main/monitor/asyncapi.yaml) and [Notifier Service](https://github.com/amadeus4dev/amadeus-async-flight-status/blob/main/notifier/asyncapi.yaml) can be found on GitHub.
+
+## Monitoring flight status information 
+
+The Monitor service checks the status of the user’s flight by calling the On-Demand Flight Status API, which provides real-time flight schedule information like departure/arrival times, gate, or terminal. A simple cURL request to the API shows how the information is represented:
+
+        curl https://test.api.amadeus.com/v2/schedule/flights?carrierCode=KL&flightNumber=1772&scheduledDepartureDate=2021-02-18 -H 'Authorization: Bearer dzh1cpJiFgAlE7iZS'
```

Comment:
```suggestion
```sh
# To get your own authorization token, follow this guide  https://developers.amadeus.com/get-started/get-started-with-self-service-apis-335

curl https://test.api.amadeus.com/v2/schedule/flights?carrierCode=KL&flightNumber=1772scheduledDepartureDate=2021-02-18 -H 'Authorization: Bearer dzh1cpJiFgAlE7iZS'
```
```